### PR TITLE
fixes #25732 - select MIME type for reports

### PR DESCRIPTION
**Pull request by amitkarsale**
Choose MIME type for report download.

### CODE REVIEW

File: app/models/report_template.rb [hunk 1]
```diff
@@ -40,8 +40,9 @@ def taxonomy_foreign_conditions
     { :report_template_id => id }
   end
 
-  def suggested_report_name
-    "#{name}-#{Date.today}.txt"
+  def suggested_report_name(type)
```
Comment (ohadlevy) [inline app/models/report_template.rb]: can write (type = 'csv') in method signature

File: app/models/report_template.rb [hunk 2]
```diff
@@ -40,8 +40,9 @@ def taxonomy_foreign_conditions
...
+    type ||= "csv"
+    "#{name}-#{Date.today}.#{type}"
```
Comment (ohadlevy) [inline app/models/report_template.rb:44]: shouldnt this change the content html header? e.g. something like 'application/csv' or similar?

**Comment by ares** [comment]
Note that if we use report_render macro, a format can be specified there, e.g. `report_render :yaml`. It would be nice if we pass the format into the renderer so that this macro could change format accordingly. For unknown formats such as txt, it would still default to csv but for yaml, it would already switch to yaml. Would that be feasible?

**Review by ares — CHANGES_REQUESTED** [review]
so two points we discussed online
1) report_render macro should look at the format and default to it, https://gist.github.com/ares/096c249fd7a1a74aebbf69ac39eaf00e
2) we need also API and hammer support for this new field
3) the format should support csv/yaml/txt, macro should keep defaulting to csv for unknown formats probably

**Comment by mmoll** [comment]
@amitkarsale please rebase :)

**Comment by ares** [comment]
The rebase didn't seem to work well. We discussed how mime-types should be implemented with a bigger team that focused on report templates recently and came to something slightly different.

We'd like to have formats defined in some constant. Each template could be assigned one or more available formats for rendering. Today we support json, yaml and csv as formats and we need to add plain text. Each format should define it's mime-type, file suffix and format constant passed to `render_report` macro. During generating, if the template has only one available format set, it will be automatically chosen, respective mime-type and suffix is used. If it has more, user can chose which format to use for the current rendering.

@amitkarsale would you be interested in updating the PR to this logic? We believe this would address all related RFEs.

**Comment by amitkarsale** [comment]
@ares , sure would implement the logic. Can we close the PR would implement in a new PR.

**Comment by ares** [comment]
sure, closing for now, thanks

**Pull request closed without merging**